StackOverflow/Model/QuestionRequest.m

```diff
@@ -0,0 +1,29 @@
+//
+//  QuestionRequest.m
+//  StackOverflow
+//
+//  Created by Malea on 2017/10/04.
+//  Copyright © 2017 DVT. All rights reserved.
+//
+
+#import "QuestionRequest.h"
+#import "DataDelegate.h"
+
+@implementation QuestionRequest
+
+- (void)fetchData {
+    
+    NSString *urlAsString = [NSString stringWithFormat:@"https://api.stackexchange.com/2.2/questions?pagesize=50&order=desc&sort=creation&tagged=ios&site=stackoverflow"];
+    
+    NSURL *url = [[NSURL alloc] initWithString:urlAsString];
+  
+    NSURLSession *session = [NSURLSession sharedSession];
+    [[session dataTaskWithURL:url completionHandler:^(NSData *data, NSURLResponse *response, NSError *error){
+        if (error) {
+            [self.delegate fetchingDataFailedWithError:error];
```

Comment:
I mentioned earlier on that you should define the methods on your delegate as required. The way this code stands it could crash because the delegate could be non nil and it could not understand these methods. There are two ways to correct this, either test to see whether the delegate actually implements the method or make the methods in the protocol required. Good practice also suggests that you should actually test to see whether the delegate is nil or not. This code will work on OC but it could crash in Swift.